### PR TITLE
Mount Point crawler lists /Volume with four variations which is confusing

### LINKED ISSUE #622: Mount Point crawler lists /Volume with four variations which is confusing
In the Mount Point crawler, how do we end up with four spellings for Volume ?
I think we should clean this up in UCX and escalate to platform.
We can also enhance this view with a action/recommendation column.

/ | DatabricksRoot |  
-- | -- | --
/Volume | DbfsReserved |  
/Volumes | UnityCatalogVolumes |  
/databricks-datasets | databricks-datasets |  
/databricks-results | databricks-results |  
/databricks/mlflow-registry | databricks/mlflow-registry |  
/databricks/mlflow-tracking | databricks/mlflow-tracking |  
/volume | DbfsReserved |  
/volumes | DbfsReserved

### PR DESCRIPTION
**Pull request by dipankarkush-db**
This pull request includes changes to the `locations.py` file in the `databricks/labs/ucx/hive_metastore` directory and the addition of a new unit test in `test_locations.py`. The changes in `locations.py` involve modifying the `_deduplicate_mounts` method in the `Mounts` class. The method now checks if the name of the `Mount` object contains the word "volume", and if so, it standardizes the name to "/Volume". This is done to ensure that mounts with different casing for the word "volume" are treated as the same mount and not duplicated in the deduplicated list.

The new unit test in `test_locations.py` verifies that the `_deduplicate_mounts` method correctly deduplicates a list of `Mount` objects with different casing for the word "volume" in their names. The test creates a list of `Mount` objects with different casing for the word "volume" and passes it to the `_deduplicate_mounts` method. The test then asserts that the deduplicated list contains only one `Mount` object with the name "/Volume". This test ensures that the changes to the `_deduplicate_mounts` method work as intended and prevent duplication of mounts with different casing for the word "volume" in their names.

In summary, the changes in this pull request modify the `_deduplicate_mounts` method to standardize the name of mounts containing the word "volume" and add a unit test to ensure that the method correctly deduplicates a list of `Mount` objects with different casing for the word "volume" in their names.

### CODE REVIEW

File: tests/unit/hive_metastore/test_locations.py [hunk 1]
```diff
@@ -46,6 +46,20 @@ def test_list_mounts_should_return_a_deduped_list_of_mount_without_encryption_ty
     assert expected == backend.rows_written_for("hive_metastore.test.mounts", "append")
 
 
+def test_list_mounts_should_return_a_deduped_list_of_mount_without_variable_volume_names():
+    mounts = [
+        Mount(name="/Volume", source="DbfsReserved"),
+        Mount(name="/Volumes", source="DbfsReserved"),
+        Mount(name="/volume", source="DbfsReserved"),
+        Mount(name="/volumes", source="DbfsReserved"),
```
Comment (nfx) [inline tests/unit/hive_metastore/test_locations.py:54]: Add normal mount and assert it appears

File: tests/unit/hive_metastore/test_locations.py [hunk 2]
```diff
@@ -46,6 +46,20 @@ def test_list_mounts_should_return_a_deduped_list_of_mount_without_encryption_ty
     assert expected == backend.rows_written_for("hive_metastore.test.mounts", "append")
 
 
+def test_list_mounts_should_return_a_deduped_list_of_mount_without_variable_volume_names():
+    mounts = [
+        Mount(name="/Volume", source="DbfsReserved"),
+        Mount(name="/Volumes", source="DbfsReserved"),
```
Comment (nfx) [inline tests/unit/hive_metastore/test_locations.py:52]: Can we just filter out dbfs reserved mounts instead?

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/unit/hive_metastore/test_locations.py (hunk 1 above)
Comment (dipankarkush-db) [reply tests/unit/hive_metastore/test_locations.py]: Done!

File: tests/unit/hive_metastore/test_locations.py (hunk 2 above)
Comment (dipankarkush-db) [reply tests/unit/hive_metastore/test_locations.py]: checked for DbfsReserved.
